compass/ocean/tests/hurricane/init/interpolate_atm_forcing.py

```diff
@@ -0,0 +1,269 @@
+
+    """
+    def __init__(self, test_case, mesh, storm):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.ocean.tests.hurricane.init.Init
+            The test case this step belongs to
+
+        mesh : compass.ocean.tests.global_ocean.mesh.Mesh
+            The test case that creates the mesh used by this test case
+
+        storm : str
+            The name of the storm to setup
+        """
+        super().__init__(test_case=test_case, name='interpolate',
+                         cores=1, min_cores=1, threads=1)
+
+        self.plot = True
+        self.plot_interval = 100
+
+        self.wind_file = 'wnd10m.nc'
+        self.pres_file = 'prmsl.nc'
+        self.forcing_file = 'atmospheric_forcing.nc'
+        self.grid_file = 'mesh.nc'
+
+        self.add_input_file(
+            filename=self.wind_file,
+            target=f'{storm}_wnd10m.nc',
+            database='initial_condition_database')
+
+        self.add_input_file(
+            filename=self.pres_file,
+            target=f'{storm}_prmsl.nc',
+            database='initial_condition_database')
+
+        mesh_path = mesh.mesh_step.path
+
+        self.add_input_file(
+            filename='mesh.nc',
+            work_dir_target=f'{mesh_path}/culled_mesh.nc')
+
+        self.add_output_file(filename=self.forcing_file)
+
+    def interpolate_data_to_grid(self, grid_file, data_file, var):
+        """
+        Interpolate time snaps of gridded data field to MPAS mesh
+        """
+
+        # Open files
+        data_nc = netCDF4.Dataset(data_file, 'r')
+        grid_nc = netCDF4.Dataset(grid_file, 'r')
+
+        # Get grid from data file
+        lon_data = data_nc.variables['lon'][:]
+        lon_data = np.append(lon_data, 360.0)
+        lat_data = np.flipud(data_nc.variables['lat'][:])
+        time = data_nc.variables['time'][:]
+        nsnaps = time.size
+        nlon = lon_data.size
+        nlat = lat_data.size
+        data = np.zeros((nsnaps, nlat, nlon))
+        print(data.shape)
+
+        # Get grid from grid file
+        lon_grid = grid_nc.variables['lonCell'][:]*180.0/np.pi
+        lat_grid = grid_nc.variables['latCell'][:]*180.0/np.pi
+        grid_points = np.column_stack((lon_grid, lat_grid))
+        ncells = lon_grid.size
+        interp_data = np.zeros((nsnaps, ncells))
+        print(interp_data.shape)
+        print(np.amin(lon_grid), np.amax(lon_grid))
+        print(np.amin(lat_grid), np.amax(lat_grid))
+
+        # Interpolate timesnaps
+        for i, t in enumerate(time):
+            print(f'Interpolating {var}: {i}')
+
+            # Get data to interpolate
+            data[i, :, 0:-1] = np.flipud(data_nc.variables[var][i, :, :])
+            data[i, :, -1] = data[i, :, 0]
+
+            interp_data[i, :] = interp_bilin(lon_data, lat_data,
+                                             data[i, :, :],
+                                             lon_grid, lat_grid)
```

Comment:
Indeed, it seems like the maximum latitude of your input data is: 89.84351
The maximum latitude of a cell center is: 89.86951361620963

It seems like the atmosphere data may be cell centered.  This function isn't smart enough to do bilinear interpolation on cell-centered data because it doesn't understand how to interpolate across the pole.  You might want to consider using `ESMF_RegridWeightGen` to generate bilinear interpolation weights and then `ncremap` to do the interpolation.    If you want to use this function, you would probably need to add a point in the input data at the pole and give it the mean value of the data in the top row of the data array.

Reply:
Another option is to add one latitude that is `180. - lat[-1]` at the end of `lat` (i.e. extrapolate beyond 90) and then copy the corresponding row of data but flip it across the pole `data[i, -1, :] = data[i, -2, ::-1]`

Reply:
That's my guess too. Thanks for the suggestion on fixing it!